### ISSUE TITLE
[CupertinoPageRoute] Gestures don't respond when clicking immediately after swiping back

### ISSUE THREAD

**Issue by yaminet1024** [flutter/flutter#73026]
**The problem is that after I use cupertionPageroute, when I swipe back to the previous page on iOS device, there is a high probability that the first time I click on the content of the previous page, nothing happens until I click twice, especially when you are halfway through the slide. Let go, and then quickly click on the previous page. The online business here is very seriously affected.**

### video

https://user-images.githubusercontent.com/23304639/103204325-11c13800-4932-11eb-93c2-a0bce2b11723.mov

### flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel unknown, 1.22.4, on macOS 11.1 20C69 darwin-x64, locale zh-Hans-CN)
[!] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses
[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
[!] Android Studio (version 4.1)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] VS Code (version 1.52.1)
[✓] Connected device (2 available)

! Doctor found issues in 2 categories.

### this is the mini example

**main.dart**

```import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:trace_startup/page.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: FlatButton(
        color: Colors.blue,
        onPressed: () {
          Navigator.push(context, CupertinoPageRoute(builder: (context) {
            return SecondPage();
          }));
        },
        child: Container(
          height: double.infinity,
          width: double.infinity,
          child: Center(
            child: Text(
              'push new page',
            ),
          ),
        ),
      ),
    );
  }
}
```

**page.dart**
```
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

class SecondPage extends StatelessWidget{
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text("second page"),),
      body: Center(child: Text("second page"),),
    );
  }

}
```

**Comment by darshankawar** [comment]
Tried your code sample on iPhoneSE simulator, but didn't notice the issue. I was able to swipe from second screen to first properly.


https://user-images.githubusercontent.com/67046386/103211520-6d89c200-492e-11eb-9859-ed39cd2d0403.mov


<details>
<summary>flutter doctor -v</summary>

```

[✓] Flutter (Channel stable, 1.22.5, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-IN)
    • Flutter version 1.22.5 at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 7891006299 (3 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.9.3

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (2 available)
    • sdk gphone x86 arm (mobile)         • emulator-5554
      • android-x86 • Android 11 (API 30) (emulator)
    • iPhone SE (2nd generation) (mobile) • 6C85835D-FBFD-4AB3-8DE8-B4FAD35E5367
      • ios         • com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)

• No issues found!


```

</details>

**Comment by yaminet1024** [comment]
> Tried your code sample on iPhoneSE simulator, but didn't notice the issue. I was able to swipe from second screen to first properly.

I think your sliding range is too large. Try to release it when sliding to the middle, or use a real machine to simulate the side-slip effect, it is very easy to appear

**Comment by yaminet1024** [comment]
@darshankawar look this Real machine example，Don't swipe  the pages to right edge with your fingers。This sliding effect is very common on large screen phones after iphone 11 pro max, so we have received a lot of feedback of this kind

https://user-images.githubusercontent.com/23304639/103214486-7b9a0b80-494b-11eb-959c-94faf192e89f.mov

**Comment by yaminet1024** [comment]
> > Tried your code sample on iPhoneSE simulator, but didn't notice the issue. I was able to swipe from second screen to first properly.
> 
> I think your sliding range is too large. Try to release it when sliding to the middle, or use a real machine to simulate the side-slip effect, it is very easy to appear。

https://user-images.githubusercontent.com/23304639/103214641-f105dc00-494b-11eb-89cd-fdd40000b28f.mov

**Comment by darshankawar** [comment]
<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 1.22.5, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-IN)
    • Flutter version 1.22.5 at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 7891006299 (3 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.9.3

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (2 available)
    • sdk gphone x86 arm (mobile)         • emulator-5554
      • android-x86 • Android 11 (API 30) (emulator)
    • iPhone SE (2nd generation) (mobile) • 6C85835D-FBFD-4AB3-8DE8-B4FAD35E5367
      • ios         • com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)

• No issues found!



```

</details>

**Comment by justinmc** [comment]
CC @xster

**Comment by chunhtai** [comment]
cc @dkwingsmt

**Comment by TahaTesser** [comment]
<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 2.0.1, on macOS 11.2.2 20D80 darwin-x64, locale en-GB)
    • Flutter version 2.0.1 at /Users/tahatesser/Code/flutter_stable
    • Framework revision c5a4b4029c (4 days ago), 2021-03-04 09:47:48 -0800
    • Engine revision 40441def69
    • Dart version 2.12.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.54.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (3 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E • ios            • iOS 14.4
    • macOS (desktop)        • macos                     • darwin-x64     • macOS 11.2.2 20D80 darwin-x64
    • Chrome (web)           • chrome                    • web-javascript • Google Chrome 88.0.4324.192

• No issues found!
```

```bash
[✓] Flutter (Channel master, 2.1.0-11.0.pre.145, on macOS 11.2.2 20D80 darwin-x64, locale en-GB)
    • Flutter version 2.1.0-11.0.pre.145 at /Users/tahatesser/Code/flutter_master
    • Framework revision a0ba646408 (3 days ago), 2021-03-05 22:09:03 -0500
    • Engine revision f751d04fa0
    • Dart version 2.13.0 (build 2.13.0-107.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.54.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (3 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E • ios            • iOS 14.4
    • macOS (desktop)        • macos                     • darwin-x64     • macOS 11.2.2 20D80 darwin-x64
    • Chrome (web)           • chrome                    • web-javascript • Google Chrome 88.0.4324.192

• No issues found!
```

</details>

**Comment by proformance** [comment]
I can confirm this is an issue. 

I've recorded a video that shows different types opening&closenings: https://vimeo.com/521522681

- First attempt I release it to the far right, notice how it's getting disposed immediately.
- Second attempt I release it to the left, pay attention to how long it takes until it gets disposed
- Third attempt, I release it to the left, now I also tap before it has been disposed, which shows how it is unresponsive. Only after it has been disposed it will be possible to open the page again.


I've modified the original minimally to make it stateful (and hence have the dispose function)



<details>
  <summary>Expand code here</summary>

```
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

class SecondPage extends StatefulWidget {
  @override
  State<StatefulWidget> createState() => SecondPageState();
}

class SecondPageState extends State<SecondPage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("second page"),
      ),
      body: Center(
        child: Text("second page"),
      ),
    );
  }

  @override
  void dispose() {
    print('disposing page');
    super.dispose();
  }
}
```

```import 'package:demoapp/page.dart';
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: FlatButton(
        color: Colors.blue,
        onPressed: () {
          print('opening page');
          Navigator.push(context, CupertinoPageRoute(builder: (context) {
            return SecondPage();
          }));
        },
        child: Container(
          height: double.infinity,
          width: double.infinity,
          child: Center(
            child: Text(
              'push new page',
            ),
          ),
        ),
      ),
    );
  }
}
```
</details>

<details>
  <summary>flutter doctor -v</summary>
[✓] Flutter (Channel beta, 1.26.0-17.8.pre, on macOS 11.2.2 20D80 darwin-x64, locale en-GB)
    • Flutter version 1.26.0-17.8.pre at /Users/mariusgrimstad/dev/flutter
    • Framework revision 044f2cf560 (13 days ago), 2021-02-24 13:02:05 -0800
    • Engine revision 042c82b02c
    • Dart version 2.12.0 (build 2.12.0-259.16.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/mariusgrimstad/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 49.0.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.53.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (2 available)
    • iPhone 12 Pro Max (mobile) • 2AB03A37-D1B2-463E-A73E-E1E16FB7124A • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • Chrome (web)               • chrome                               • web-javascript •
      Google Chrome 88.0.4324.192

• No issues found!
</details>

**Comment by proformance** [comment]
Looking at it more in detail it seems like animation curve combined with the inner workings of route might be the problem here.

https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/cupertino/route.dart#L735

Currently the dismissal animation is using the curve `Curves.fastLinearToSlowEaseIn`

I added some debug traces to see the animation-value and timestamps, here's a snippet how it looks close to the end of the animation.

```
...
flutter: buildTransitions 0.013482164725324042 (1618342815382) <---  barely visible as only 1% of the width is shown
flutter: buildTransitions 0.010625395290166484 (1618342815399)
flutter: buildTransitions 0.008203684573317105 (1618342815415)
flutter: buildTransitions 0.006376271156842028 (1618342815432)
flutter: buildTransitions 0.004842631100606232 (1618342815448)
flutter: buildTransitions 0.003644510584527083 (1618342815465)
flutter: buildTransitions 0.0027178166513320834 (1618342815482)
flutter: buildTransitions 0.0019632494060380923 (1618342815498)
flutter: buildTransitions 0.001398682602685053 (1618342815515)
flutter: buildTransitions 0.0009542743748859328 (1618342815532)
flutter: buildTransitions 0.0006366329552504046 (1618342815548)
flutter: buildTransitions 0.00039868855001279346 (1618342815565)
flutter: buildTransitions 0.00023982897851249074 (1618342815582)
flutter: buildTransitions 0.00013001372356491459 (1618342815599)
flutter: buildTransitions 0.00006018844980637539 (1618342815615)
flutter: buildTransitions 0.000023578998342599355 (1618342815632)
flutter: buildTransitions 0.000005529384273339488 (1618342815648)
flutter: buildTransitions 4.4712767222065963e-7 (1618342815665)   <--- animation will now stop as it reached ~0
flutter: AnimationStatusListener->AnimationStatus.dismissed (1618342815684)  <--- status has changed, will trigger pop
```

As you can see, the value is almost 0 ( == barely visible at the very right edge) for about 300ms before the AnimationStatus is equal to dismissed. This means it will look like it is popped, but is still there.

Only when the status has changed to AnimationStatus.dismissed, it will call `stopUserGesture`
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/cupertino/route.dart#L773

Digging further it turns out stopGesture actually blocks user-input.
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/routes.dart#L832

This means it will by the current design block any input for some time (even though the user is not seeing anything of the previous modal) before the user can interact again. The time will depend on how far to the right you drop it, so if you drop it far to the right it will barely affect the user, but if you make a quick swipe-to-dismiss on the left side on the screen, the user will be heavily affected.

**Comment by proformance** [comment]
@chunhtai @TahaTesser @gspencergoog 

I found that [this issue](https://github.com/flutter/flutter/issues/48903) was the reason for blocking the user-input during the userGesture. I tried to remove [this line](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/routes.dart#L773) and can confirm the feeling was much smoother as a new tap on the previous page would respond immediately. (However with some side-effects afterwards of course). Do you have any suggestion on how to solve this in another way where it doesn't have to be blocked?   

Currently the feeling is not smooth as some taps are not responding, specially when using an actual device.

**Comment by mm-public-tx** [comment]
@xster

**Comment by mm-public-tx** [comment]
Is there a schedule to solve this problem?

**Comment by doseeing** [comment]
thanks for @proformance modification, problem sovled(but maybe other things break). 

For anyone can't find related line in`https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/routes.dart` 

``` dart
// before modification
  bool get _shouldIgnoreFocusRequest {
    return widget.route.animation?.status == AnimationStatus.reverse ||
      (widget.route.navigator?.userGestureInProgress ?? false);
  }
// after modification
  bool get _shouldIgnoreFocusRequest {
    return widget.route.animation?.status == AnimationStatus.reverse; // ||
      //(widget.route.navigator?.userGestureInProgress ?? false);
  }
```

**Comment by willlockwood** [comment]
Just found this issue, after doing a bit of work to improve the fidelity to the iOS behavior. I believe that the core issue here will be resolved once my PR (https://github.com/flutter/flutter/pull/95757) lands, but would appreciate any eyes from folks here on whether there's anything I've missed?

**Comment by nt4f04uNd** [comment]
With https://github.com/flutter/flutter/issues/54090#issuecomment-609908042 and @willlockwood's PR it seems it then would be working as intended.

/cc @chunhtai should we repurpose this bug into documentation request?

**Comment by 2e2ee95304418f96** [comment]
So is there any news about the issue?

**Comment by justinmc** [comment]
@darshankawar Is this still reproducible on master now that https://github.com/flutter/flutter/pull/95757 has been merged?

CC @MitchellGoodwin

**Comment by nt4f04uNd** [comment]
@justinmc #95757 was reverted shortly after the merge

**Comment by darshankawar** [comment]
> @darshankawar Is this still reproducible on master now that #95757 has been merged?

I am unable to replicate this on iOS device and simulator. The swipe works properly on first tap:


https://user-images.githubusercontent.com/67046386/192947693-7983b733-1d54-4635-ba31-1d9aef372ec4.mov


https://user-images.githubusercontent.com/67046386/192947781-d696e372-bbb9-46d9-ab26-3a998929c9d9.MP4

<details>
<summary> runnable code sample </summary>

```

import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';


void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: ElevatedButton(
      //  color: Colors.blue,
        onPressed: () {
          Navigator.push(context, CupertinoPageRoute(builder: (context) {
            return SecondPage();
          }));
        },
        child: Container(
          height: double.infinity,
          width: double.infinity,
          child: Center(
            child: Text(
              'push new page',
            ),
          ),
        ),
      ),
    );
  }
}

class SecondPage extends StatelessWidget{
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text("second page"),),
      body: Center(child: Text("second page"),),
    );
  }

}

```

</details>

<details>
<summary> master flutter doctor -v </summary>

```

dhs@Dhss-MacBook-Pro ~ % flutter doctor -v
[✓] Flutter (Channel master, 3.4.0-34.0.pre.7, on macOS 12.2.1 21D62 darwin-x64,
    locale en-GB)
    • Flutter version 3.4.0-34.0.pre.7 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f22f1344bd (37 minutes ago), 2022-09-29 00:13:42 -0400
    • Engine revision c003042e52
    • Dart version 2.19.0 (build 2.19.0-252.0.dev)
    • DevTools version 2.17.0

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 65.1.4
    • Dart plugin version 213.7228

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 12.2.1 21D62 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 105.0.5195.125

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

</details>

**Comment by justinmc** [comment]
@darshankawar Thanks for checking!

It's strange that this isn't reproducible anymore if https://github.com/flutter/flutter/pull/95757 was reverted...  Can anyone else still reproduce this on master?  If not then I guess we can close this.

**Comment by ConProgramming** [comment]
Doesn't look to be a "first click" issue, more of a "fast click" issue. If you flick the CupertinoPageRoute closed from the left, it'll move across the screen pretty quickly and look gone, but still take some time to dispose the page widget. Only after it is disposed does the background become responsive.

The example by @darshankawar doesn't show the issue because he clicks the background pretty slowly in comparison to how a real user would interact. If you quickly swipe a route closed, then click, the issue is still reproducible.

**Comment by justinmc** [comment]
Thanks for the response, let's keep this open as something that should be fixed then.

**Comment by TahaTesser** [comment]
I can confirm the issue is very much reproducible, it's not just iOS. 

<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/cupertino.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const CupertinoApp(
      theme: CupertinoThemeData(brightness: Brightness.light),
      home: FirstPage(),
    );
  }
}

class FirstPage extends StatelessWidget {
  const FirstPage({super.key});

  @override
  Widget build(BuildContext context) {
    return CupertinoPageScaffold(
      navigationBar: const CupertinoNavigationBar(
        middle: Text('FirstPage'),
      ),
      child: Center(
        child: CupertinoButton(
            onPressed: () {
              Navigator.push(
                context,
                CupertinoPageRoute<Widget>(
                  builder: (_) => const SecondPage(),
                ),
              );
            },
            child: const Text('Go to SecondPage')),
      ),
    );
  }
}

class SecondPage extends StatelessWidget {
  const SecondPage({super.key});

  @override
  Widget build(BuildContext context) {
    return CupertinoPageScaffold(
      navigationBar: const CupertinoNavigationBar(
        middle: Text('SecondPage'),
      ),
      child: Center(
        child: CupertinoButton(
          onPressed: () {
            Navigator.pop(context);
          },
          child: const Text('Pop'),
        ),
      ),
    );
  }
}

``` 
	
</details>

### Steps
1. Go to the second page by pressing the button on the first page.
2. Swipe back to go back and immediately click the button on the first page.

 I also included the `Navigator.pop` button on the second page, the issue doesn't reproduce when calling this instead of swiping back

### Results
The button on the first page won't respond for a short time.


https://user-images.githubusercontent.com/48603081/194162130-d7dc2670-e2dc-454b-96b8-5e28380bdcbf.mov

https://user-images.githubusercontent.com/48603081/194162145-3e6d72a9-4468-464f-9f88-2e50a3b3d3ea.mov

**Comment by ConProgramming** [comment]
> thanks for @proformance modification, problem sovled(but maybe other things break).
> 
> For anyone can't find related line in`https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/routes.dart`
> 
> ```dart
> // before modification
>   bool get _shouldIgnoreFocusRequest {
>     return widget.route.animation?.status == AnimationStatus.reverse ||
>       (widget.route.navigator?.userGestureInProgress ?? false);
>   }
> // after modification
>   bool get _shouldIgnoreFocusRequest {
>     return widget.route.animation?.status == AnimationStatus.reverse; // ||
>       //(widget.route.navigator?.userGestureInProgress ?? false);
>   }
> ```

I believe it's not CupertinoPageRoute specific either, because this fixed the issue for me.

**Comment by ConProgramming** [comment]
To be clear: Still very much an issue. The patch above made it usable, but, for example when opening a different route while the current one is closing, there's a weird jump in the animation. Where it previously wouldn't have responded at all it now does respond, but with a weird animation jump

**Comment by TahaTesser** [comment]
> To be clear: Still very much an issue. The patch above made it usable, but, for example when opening a different route while the current one is closing, there's a weird jump in the animation. Where it previously wouldn't have responded at all it now does respond, but with a weird animation jump

That's a great observation, given the sample reproduces the issue when swiping back with `CupertinoPageRoute`, this description is useful for anyone running into this issue or when trying to fix it.

**Comment by FantaZZ** [comment]
Has any news ?

**Comment by MitchellGoodwin** [comment]
Sorry we haven't gotten to this yet. This isn't actively being worked on yet, but it is high on my list once I finish a few other issues, unless someone else gets to it first.

**Comment by FantaZZ** [comment]
I make this as a temporary solution try fix this bug.

https://gist.github.com/FantaZZ/5a568ef74630196e32b3b461141e1360

It was copy from ```CupertinoRouteTransitionMixin.buildPageTransitions``` basically.

Only change this https://gist.github.com/FantaZZ/5a568ef74630196e32b3b461141e1360#file-custom_cupertino_page_route_builder-dart-L275

I holp it is useful for someone.

**Comment by xhzq233** [comment]
any updates?

**Comment by MitchellGoodwin** [comment]
I'm not currently working on this, but I'm leaving it assigned as this, and other route transition issues are immediate on my list after I finish the current larger issues on my queue. If someone wants to take a crack at it before then, feel free to submit a PR or reach out.

**Comment by sjamr10** [comment]
I'm facing the same issue with MaterialPageRoute, if you wait a little bit it works fine but if you tap something immediately after going back it doesn't work.

**Comment by victorsanni** [comment]
@yiiim does https://github.com/flutter/flutter/pull/153765 also fix this?

**Comment by yiiim** [comment]
> @yiiim does #153765 also fix this? Is this a dup of #48225?

Yes, the same issue.
